Src/Newtonsoft.Json/Linq/JArray.cs

```diff
@@ -93,7 +93,7 @@ internal override bool DeepEquals(JToken node)
             return (node is JArray t && ContentsEqual(t));
         }
 
-        internal override JToken CloneToken()
+        internal override JToken CloneToken(JsonCloneSettings? settings = null)
         {
             return new JArray(this);
         }
```

Comment:
It seems odd to me that even when `settings` were provided, this method will not be using it. 
Is it because the clone settings won't apply to `JArray` or because they don't apply to the scenario we are trying to solve which seems specific to `JValue`?

Reply:
If we think this is something that should be defined at `JToken` level, then I would imagine every implementation of `JToken` should make use of it. Otherwise, we should just introduce an new overload in `JValue`.

Reply:
Thanks for catching this. I included the new settings object within all JToken implementations.